backend/src/sms/routes/tests/sms-callback.routes.test.ts

```diff
@@ -0,0 +1,168 @@
+  credential = await Credential.create({ name: 'twilio' } as Credential)
+  await UserCredential.create({
+    label: `twilio-callback-${userId}`,
+    type: ChannelType.SMS,
+    credName: credential.name,
+    userId,
+  } as UserCredential)
+})
+
+beforeAll(async () => {
+  sequelize = await sequelizeLoader(process.env.JEST_WORKER_ID || '1')
+})
+
+afterEach(async () => {
+  jest.clearAllMocks()
+  await SmsMessageTransactional.destroy({ where: {} })
+  await User.destroy({ where: {} })
+  await UserCredential.destroy({ where: {} })
+  await Credential.destroy({ where: {} })
+})
+
+afterAll(async () => {
+  await sequelize.close()
+  await (app as any).cleanup()
+})
+
+describe('On successful message send, status should update according to Twilio response', () => {
+  const validApiCall = {
+    body: 'Hello world',
+    recipient: '98765432',
+    label: 'twilio-callback-1',
+  }
+  test('Should send a message successfully', async () => {
+    const mockSendMessageResolvedValue = 'message_id_callback'
+    const mockSendMessage = jest
+      .spyOn(SmsService, 'sendMessage')
+      .mockResolvedValue(mockSendMessageResolvedValue)
+    mockSecretsManager.getSecretValue().promise.mockResolvedValueOnce({
+      SecretString: JSON.stringify(TEST_TWILIO_CREDENTIALS),
+    })
+    const res = await request(app)
+      .post('/transactional/sms/send')
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send(validApiCall)
+
+    expect(res.status).toBe(201)
+    expect(mockSendMessage).toBeCalledTimes(1)
+
+    const transactionalSms = await SmsMessageTransactional.findOne({
+      where: { userId: user.id.toString() },
+      order: [['createdAt', 'DESC']],
+    })
+    const transactionalSmsId = transactionalSms?.id
+
+    const getByIdRes = await request(app)
+      .get(`/transactional/sms/${transactionalSmsId}`)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send()
+    expect(getByIdRes.status).toBe(200)
+    expect(getByIdRes.body.status).toBe(TransactionalSmsMessageStatus.Unsent)
+    expect(getByIdRes.body.body).toEqual('Hello world')
+    expect(getByIdRes.body.recipient).toEqual('98765432')
+    expect(getByIdRes.body.credentialsLabel).toEqual('twilio-callback-1')
+
+    expect(getByIdRes.status).toBe(200)
+    expect(getByIdRes.body.status).toBe(TransactionalSmsMessageStatus.Unsent)
+    const sampleTwilioCallback = {
+      SmsSid: mockSendMessageResolvedValue,
+      SmsStatus: 'sent',
+      MessageStatus: 'sent',
+      To: '+1512zzzyyyy',
+      MessageSid: mockSendMessageResolvedValue,
+      AccountSid: 'ACxxxxxxx',
+      From: '+1512xxxyyyy',
+      ApiVersion: '2010-04-01',
+      ErrorCode: 'ERRORBOI',
+    }
+    sampleTwilioCallback.ErrorCode = ''
+
+    jest
+      .spyOn(SmsCallbackService, 'isAuthenticatedTransactional')
+      .mockReturnValue(true)
+    let callbackRes = await request(app)
+      .post('/callback/sms')
+      .set('Authorization', `Basic ${1234}`)
```

Comment:
**[nit]** can we abstract away the callback credentials? `Basic ${1234}` looks weird haha

Reply:
it needs to be basic auth otherwise it'll reject the request, i'll change it to something more readable

Reply:
Ohh ya I meant put 1234 as a variable. Previously 1234 was in backticks and `${}` but it's not a variable. 

So right now (after your latest commit), it's still a bit weird since we're using backticks but there are no variables. 

Just a nit!

Reply:
ohh ok i get it, updated